### PR TITLE
Update PR Reviewer Checklist with auto-merge

### PR DESCRIPTION
**Pull request by jackkoenig**
Inform/Remind reviewers to clean up commit messages

### Contributor Checklist

- [NA] Did you add Scaladoc to every public function/method?
- [NA] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [NA] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [NA] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

 - documentation
 
#### API Impact

No impact

#### Backend Code Generation Impact

No impact

#### Desired Merge Strategy

   - Squash

### Reviewer Checklist (only modified by reviewer)
- [x] Did you add the appropriate labels?
- [x] Did you mark the proper milestone (Bug fix: `3.4.x`, [small] API extension: `3.5.x`, API modification or big change: `3.6.0`)?
- [x] Did you review?
- [x] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mwachs5) [inline .github/PULL_REQUEST_TEMPLATE.md:51]: can we put this before the "did you mark as `Please Merge`"? I feel like that should be the last one...

I forget how our merge rules work. will GitHub's auto-merge NOT merge unless you've labeled with `Please Merge`? If not, we should probably just get rid of the `Please Merge` label and automation...

Comment (jackkoenig) [reply .github/PULL_REQUEST_TEMPLATE.md:51]: Github's auto-merge will merge even without `Please Merge` but only if the PR is up-to-date. We still need Mergify to merge master to get PRs up-to-date.

Comment (mwachs5) [reply .github/PULL_REQUEST_TEMPLATE.md:50]: so does `Please Merge` automatically merge master?

Comment (mwachs5) [reply .github/PULL_REQUEST_TEMPLATE.md:50]: basically i think this checklist is a bit confusing, maybe we should add a header "WHEN YOU REALLY WANT TO MERGE IT" then those last two checklist boxes, in swapped order?

Comment (mwachs5) [reply .github/PULL_REQUEST_TEMPLATE.md:50]: (you already swapped the order, the order is good now...)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mwachs5) [inline .github/PULL_REQUEST_TEMPLATE.md:50]:
```suggestion
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash) and clean up the commit message
  - [ ] Merge: Label with `Please Merge`
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mwachs5) [reply .github/PULL_REQUEST_TEMPLATE.md:50]: i made a suggest change.... it seems like we have 2 divergent paths for Merge vs Squash commit now.

Comment (jackkoenig) [reply .github/PULL_REQUEST_TEMPLATE.md]: Technically they've always been divergent but not clearly captured there because if you're doing the "Rebase" mode, the contributor is supposed to rebase to clean up the history and update it to master (no `Please Merge`), but yeah this should be made more clear

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mwachs5) [inline .github/PULL_REQUEST_TEMPLATE.md:53]:
```suggestion
```

**Comment by mwachs5** [comment]
i pushed a commit to remove the redundant line. LGTM!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mwachs5) [inline .github/PULL_REQUEST_TEMPLATE.md:51]:
```suggestion
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)` and clean up the commit message
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (jackkoenig) [inline .github/PULL_REQUEST_TEMPLATE.md:52]: Now that I think about it, this isn't actually right.

For squash, we do **both** Enable `auto-merge (squash)` and label with `Please Merge`. For rebase+merge, the contributor is supposed to manually update it and then the reviewer (or contributor if they have write access) are supposed to merge with merge commit.

**Comment by sequencer** [comment]
Just checked the mergify documentation, seems we are using the “rebase” function from mergify, and GitHub Auto Merge triggers faster than mergify.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mwachs5) [reply .github/PULL_REQUEST_TEMPLATE.md]: i edited this please take a look

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (jackkoenig) [inline .github/PULL_REQUEST_TEMPLATE.md:52]:
```suggestion
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.
```
`Please Merge` only does squash-and-merge. Getting a merge commit requires manually merging

**Review by mwachs5 — APPROVED** [review]

**Comment by mwachs5** [comment]
Please use the auto-merge to clean up your commit history ;)

**Comment by mwachs5** [comment]
Maybe we should change the `Please Merge` label to "Please Squash and Merge" and unclear why we even have the label if GitHub now has the auto-squash

**Comment by jackkoenig** [comment]
> Maybe we should change the Please Merge label to "Please Squash and Merge" and unclear why we even have the label if GitHub now has the auto-squash

Good point, the main thing Mergify is doing here is creating the "Merge Queue" and updating the PRs to current master. Perhaps we can tweak how we use Mergify to not need the label but instead notice Auto-merge is enabled, or perhaps to not merge themselves but only let Github Auto-merge do the merging...

**Pull request merged**
